site/_docs/howto.md

```diff
@@ -806,7 +806,7 @@ git clean -xn
 ./gradlew prepareVote -Prc=0
 
 # Push release candidate to ASF servers
-./gradlew prepareVote -Prc=0 -Pasf -Pasf.git.pushRepositoryProvider=GITBOX
```

Comment:
Throws following exception with `GITBOX`:
```
> Task :pushRcTag FAILED

Build calcite FAILURE reason:
    Execution failed for task ':pushRcTag':
        Caused by: org.eclipse.jgit.api.errors.TransportException: https://gitbox.apache.org/repos/asf/calcite.git: not authorized
            at org.eclipse.jgit.api.PushCommand.call(PushCommand.java:147)
            at com.github.vlsi.gradle.release.jgit.dsl.GitExtensionsKt.push(GitExtensions.kt:132)
            at com.github.vlsi.gradle.release.GitPushTask$pushTag$1.invoke(GitPushTask.kt:54)
            at com.github.vlsi.gradle.release.GitPushTask$pushTag$1.invoke(GitPushTask.kt:30)
            at com.github.vlsi.gradle.release.DefaultGitTask.jgit(DefaultGitTask.kt:45)
            at com.github.vlsi.gradle.release.GitPushTask.pushTag(GitPushTask.kt:50)
```

Reply:
For avatica, I also have `-Pasf.git.pushRepositoryProvider=GITBOX` in the release script: https://github.com/apache/calcite-avatica/blob/main/docker.sh#L298

I haven't used this recently, so not sure if it still works.

Reply:
CC @NobiGo @zabetak since you are the latest release managers.

Reply:
The `asf.git.pushRepositoryProvider` should be inline with `asfGitSourceUsername` and `asfGitSourcePassword`. If you are using `GITBOX` then you should use the ASF credentials; if you use `GITHUB` you should use the GitHub credentials. I assume both providers should work fine if the correct credentials are set.

I think there were some misconception that GITHUB was not working in the past (https://issues.apache.org/jira/browse/CALCITE-4856) but I suspect that the underlying reason was the same (mixing ASF vs. GitHub credentials).

If we want to improve something it makes sense to clarify somewhere this story about the different credentials and not just change again the default values.

Reply:
I was following this line to setup the environment: https://github.com/apache/calcite/blob/816edd1fa1a1d6af7e72416d791eb01d8c66b6ea/site/_docs/howto.md?plain=1#L686-L687

I thought this is the suggested way to interact with Git. (Actually I'm not sure how to setup Gitbox credentials, I guess `asfGitSourcePassword` is the apache password which is the same with `asfNexusPassword` and
`asfSvnPassword`?)

Reply:
Yes, the Gitbox credentials is the usual username and password combination that is used to access most (if not all) ASF services.

Reply:
I'll leave the default as 'GITBOX', and add more comments about it.